suite-native/device/src/hooks/useDeviceAuthenticityCheck.tsx

```diff
@@ -0,0 +1,199 @@
+import { useCallback } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { useNavigation } from '@react-navigation/native';
+
+import {
+    CheckDeviceAuthenticityThunkResult,
+    deviceAuthenticityActions,
+} from '@suite-common/device-authenticity';
+import { selectSelectedDevice } from '@suite-common/wallet-core';
+import { DeviceAuthenticityCheckResult, EventType, analytics } from '@suite-native/analytics';
+import { requestPrioritizedDeviceAccess } from '@suite-native/device-mutex';
+import { FeatureFlag, useFeatureFlag } from '@suite-native/feature-flags';
+import { Translation } from '@suite-native/intl';
+import { captureSentryMessage, withSentryScope } from '@suite-native/sentry';
+import { useToast } from '@suite-native/toasts';
+import TrezorConnect from '@trezor/connect';
+
+export const useDeviceAuthenticityCheck = () => {
+    const navigation = useNavigation();
+    const dispatch = useDispatch();
+    const { showToast } = useToast();
+    const allowDebugKeys = useFeatureFlag(FeatureFlag.IsDebugKeysAllowed);
+
+    const device = useSelector(selectSelectedDevice);
+    const isDeviceBootloaderUnlocked = !!device && !device?.features?.bootloader_locked;
+    const reportCheckResult = useCallback(
+        (
+            result: DeviceAuthenticityCheckResult,
+            error?: string,
+            payload?: CheckDeviceAuthenticityThunkResult,
+        ) => {
+            analytics.report({
+                type: EventType.DeviceSettingsAuthenticityCheck,
```

Comment:
Reusing the EventType from Device Settings, even for the Onboarding, should not be a problem for us. We can always check in analytics from what screen it was reported based on other events, if needed. But Sentry reports will be more practical for us.